activation/activation.go

```diff
@@ -346,35 +348,143 @@ func (b *Builder) buildInitialPost(ctx context.Context, nodeID types.NodeID) err
 	}, postInfo.NumUnits)
 	if err != nil {
 		b.log.Error("initial POST is invalid", log.ZShortStringer("smesherID", nodeID), zap.Error(err))
-		if err := nipost.RemoveInitialPost(b.localDB, nodeID); err != nil {
-			b.log.Fatal("failed to remove initial post", log.ZShortStringer("smesherID", nodeID), zap.Error(err))
-		}
-		return fmt.Errorf("initial POST is invalid: %w", err)
+		return nil, nil, fmt.Errorf("initial POST is invalid: %w", err)
+	}
+
+	if err := nipost.AddPost(b.localDB, nodeID, initialPost); err != nil {
+		b.log.Error("failed to save initial post", zap.Error(err))
 	}
 
 	metrics.PostDuration.Set(float64(time.Since(startTime).Nanoseconds()))
 	public.PostSeconds.Set(float64(time.Since(startTime)))
 	b.log.Info("created the initial post")
+	return post, postInfo, nil
+}
 
-	return nipost.AddInitialPost(b.localDB, nodeID, initialPost)
+// Obtain certificates for the poets.
+// We want to certify immediately after the startup or creating the initial POST
+// to avoid all nodes spamming the certifier at the same time when
+// submitting to the poets.
+func (b *Builder) certifyPost(ctx context.Context, post *types.Post, meta *types.PostInfo, ch []byte) {
+	client := NewCertifierClient(b.log, post, meta, ch, WithCertifierClientConfig(b.certifierConfig.Client))
+	certifier := NewCertifier(b.localDB, b.log, client)
+	certifier.CertifyAll(ctx, b.poets)
+
+	b.smeshingMutex.Lock()
+	b.certifiers[meta.NodeID] = certifier
+	b.smeshingMutex.Unlock()
 }
 
-func (b *Builder) run(ctx context.Context, sig *signing.EdSigner) {
-	defer b.log.Info("atx builder stopped")
+func (b *Builder) obtainPostFromLastAtx(
+	ctx context.Context,
+	nodeId types.NodeID,
+) (*types.Post, *types.PostInfo, []byte, error) {
```

Comment:
Same here, we already have a type that combines all the relevant information of a post in one type:

```suggestion
func (b *Builder) obtainPostFromLastAtx(ctx context.Context, nodeId types.NodeID) (*nipost.Post, error) {
```